cmk/special_agents/agent_cisco_meraki.py

```diff
@@ -376,23 +455,26 @@ def parse_arguments(argv: Sequence[str] | None) -> argparse.Namespace:
         default=[],
         help="Explicit organisation IDs that are checked.",
     )
-
     return parser.parse_args(argv)
 
 
-@dataclass(frozen=True)
-class MerakiConfig:
-    dashboard: meraki.DashboardAPI
-    hostname: str
-    section_names: Sequence[str]
-
-
 def _get_organisations(config: MerakiConfig, org_ids: Sequence[str]) -> Sequence[_Organisation]:
     if not _need_organisations(config.section_names):
         return []
-    return (
-        GetOrganisationsByIDCache(config, org_ids) if org_ids else GetOrganisationsCache(config)
-    ).get_live_data()
+    organisations = [
```

Comment:
the idea of configured IDs is the following:

- If you configure IDs then - for these IDs - data is fetched (from API or from cache) to enrich the organisation data
- If not then collect all IDs (from API or from cache)

Here you mix up these two concepts.